mailroom file with classes and test file

colephalen opened the pull request.

colephalen committed:
mailroom file with classes and test file

colephalen committed:
adding missing dict and set lab

nairraghav commented on students/ColeP/Session09/mailroom_oo.py:
```diff
@@ -0,0 +1,185 @@
+          }
+
+
+class Donor:
```
Generally, we want to pull out our models into their own files

nairraghav commented on students/ColeP/Session09/mailroom_oo.py:
```diff
@@ -0,0 +1,185 @@
+        return name.lower().strip().replace(' ', '')
+
+
+class StatsTable:
```
I'm a little skeptical that you need to create this class just for the stats, you should be able to do everything within the donor class itself

nairraghav commented on students/ColeP/Session09/mailroom_oo.py:
```diff
@@ -0,0 +1,185 @@
+#         return d
+
+
+def main_menu():
```
Don't forget to actually implement your OO changes in this version of mailroom! All your methods should be using the donor models instead of this default donors dict.

nairraghav commented:
I've given credit for the dict lab. 

It looks you have a good start on the OO mailroom assignment, don't forget to actually use the donor models in your mailroom! Please submit those changes in a new PR

colephalen commented:
Hi Ron,

Couple questions. What do you mean "actually use the donor models in your
mailroom"? I cleaned up the file a bit (sorry i submitted work with all
those comments) and will resubmit. I also changed main_menu so that my
dictionary is built using class methods.

I'll also try to incorporate the functionality of my second stats class
into the donor class.

Some grading q's:
Can you comment on why I haven't received credit for the fibonacci sequence
exercise and mailroom, pt2? I submitted fib.py a while ago and when working
on a later version of mailroom i incorporated file writing and dictionary
use under the impression that i would receive credit for mailroom 2.

If I complete half of the HTML renderer assignment will i receive half
credit?

Thanks!
Cole



On Thu, Jun 13, 2019 at 5:28 PM nairraghav <notifications@github.com> wrote:

> Merged #180
> <https://github.com/UWPCE-PythonCert-ClassRepos/SP_2019_210A_classroom/pull/180>
> into master.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/UWPCE-PythonCert-ClassRepos/SP_2019_210A_classroom/pull/180?email_source=notifications&email_token=ALESKBFTLRUDD5LJA36KIKDP2LQZVA5CNFSM4HXEN5NKYY3PNVWWK3TUL52HS4DFWZEXG43VMVCXMZLOORHG65DJMZUWGYLUNFXW5KTDN5WW2ZLOORPWSZGOR7EB4CY#event-2412256779>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALESKBFWCPPLCRO3PPJVDOTP2LQZVANCNFSM4HXEN5NA>
> .
>

nairraghav commented:
Hey Cole,

The main goal of the Mailroom OO assignment is to take advantage of the Donor / DonorCollection classes that you created. These classes should handle the logic of the donors / donorcollection. For example, when adding a new donor to mailroom:
1. Use the donor __init__ to create a Donor object
2. Use the donor_collection to insert this Donor to keep track of all donors that exist

The reason you did not receive credit for fib.py was because you did not implement the functions "lucas" and and the "sum_series". I can give some partial credit so that you have something in place for now. 

I never saw a mailroom2 file so i had't given credit but you make a good point. You have implemented the changes involved in it when doing Mailroom3. I will go ahead and give you credit for that!

Yes! You will receive credit for HTMLRender for what you've done. Please let me know when you feel that you have the final submission for it and I will give the appropriate credit for it.

Thanks,
Ron
